virtio-devices/src/transport/pci_device.rs

```diff
@@ -882,11 +882,20 @@ impl PciDevice for VirtioPciDevice {
         let device = device_clone.lock().unwrap();
 
         let mut settings_bar_addr = None;
+        let mut use_64bit_bar = self.use_64bit_bar;
```

Comment:
I hate that you have to do this. Could you instead intialise this value correctly at the point at which you initially set it (like `set_state()` rather than doing it here.)

Reply:
I guess you're talking about `self.use_64bit_bar`, right? Because `use_64bit_bar` is a local variable. The problem is that `set_state()` would happen too late.